### PR TITLE
[US][Structure][Main][ProductCategoriesPage] Generic type for the ProductCategoriesPage<AleksandrAkimovChg>

### PR DESCRIPTION
**Pull request by AleksandrAkimovChg**
change ProductCategoryPage to ProductsPage and added generic type for ProductsPage 

[US][Structure] - https://trello.com/c/90HzB1nK/21-usstructuremainproductcategoriespage-generic-type-for-the-productspagealeksandrakimovchg

### CODE REVIEW

**Comment by AleksandrAkimovChg** [comment]
Please review change.

**Review by maksymiasko — APPROVED** [review]

File: src/test/java/pages/base_abstract/MainPage.java [hunk 1]
```diff
@@ -22,59 +24,64 @@ public abstract class MainPage extends BasePage {
 //    breadcrumbs
 
     private static final String CATEGORYMENU = "//section[@id='categorymenu']";
+    private static final String NAV_PILLS_CATEGORYMENU = CATEGORYMENU + "/nav/ul/li";
+
     private static final String CONTAINS = "//a[contains(text(),";
```
Comment (IraKalinka) [inline src/test/java/pages/base_abstract/MainPage.java:29]: private static final String HREF = "//a[@href='https://automationteststore.com/index.php?rt=product/category&path=";

File: src/test/java/pages/base_abstract/MainPage.java [hunk 2]
```diff
@@ -22,59 +24,64 @@ public abstract class MainPage extends BasePage {
...
 
-    @FindBy(xpath = CATEGORYMENU + CONTAINS + "'Apparel & accessories')]")
+    @FindBy(css = "#categorymenu > nav > ul > li:nth-child(2) > a")
     private WebElement apparelAccessoriesMenu;
 
-    @FindBy(xpath = CATEGORYMENU + CONTAINS + "'Makeup')]")
+    @FindBy(css = "#categorymenu > nav > ul > li:nth-child(3) > a")
     private WebElement makeupMenu;
 
     @FindBy(xpath = CATEGORYMENU + CONTAINS + "'Cheeks')]")
```
Comment (IraKalinka) [inline src/test/java/pages/base_abstract/MainPage.java:41]: @FindBy(xpath = CATEGORYMENU + HREF + "36_40']")

File: src/test/java/pages/base_abstract/MainPage.java (hunk 1 above)
Comment (IraKalinka) [reply src/test/java/pages/base_abstract/MainPage.java:29]: это вместо CONTAINS

File: src/test/java/pages/base_abstract/MainPage.java (hunk 2 above)
Comment (IraKalinka) [reply src/test/java/pages/base_abstract/MainPage.java:41]: это новые Щёки

**Pull request merged**